From 4e152fab4a23b50f510c3a15cc600f75ac6967a0 Mon Sep 17 00:00:00 2001
From: chen xu <fionaxu@google.com>
Date: Tue, 15 Jan 2019 12:09:50 -0800
Subject: [PATCH 25/39] Update sharedPreference to false when exits new device
 provisioning

Update sharedPreference to false when exits new device provisioning,
indicating no users modifications on the settings for new devices.
Thus carrier specific default roaming settings can be applied for new devices.

Bug: 122817867
Test: Manual test on VZW SIM in Setupwizard
Change-Id: Ibe4081f863095bca44d761dbfc3e5741a039c8e6
Merged-in: I8742c78ea99679ec744bf3bf82b210b5de114a45
---
 .../internal/telephony/dataconnection/DcTracker.java      | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index fc9dd8609..1c4e7f478 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -864,6 +864,14 @@ public class DcTracker extends Handler {
         } else {
             onCleanUpAllConnections(Phone.REASON_DATA_SPECIFIC_DISABLED);
         }
+        // Update sharedPreference to false when exits new device provisioning, indicating no users
+        // modifications on the settings for new devices. Thus carrier specific
+        // default roaming settings can be applied for new devices till user modification.
+        final SharedPreferences sp = PreferenceManager
+                .getDefaultSharedPreferences(mPhone.getContext());
+        if (!sp.contains(Phone.DATA_ROAMING_IS_USER_SETTING_KEY)) {
+            sp.edit().putBoolean(Phone.DATA_ROAMING_IS_USER_SETTING_KEY, false).commit();
+        }
     }
 
 
-- 
2.17.1

